### PR TITLE
UI: Migrate preview deployments from Netlify to Vercel

### PR DESCRIPTION
**Pull request by backspace**
This adds:

* a script for building and deploying the Ember UI and Storybook to
  Vercel
* configuration for that deployment
* a header link to the UI to link to Storybook when built with
  `STORYBOOK_LINK=true`

image

It also removes a file used to configure Netlify redirects.

The Netlify setup had two “sites”: `nomad-storybook` and `nomad-ui`. I
attempted to replicate that here but ran into some platform limitations
with Vercel: two “projects” cannot share the same root directory without
also sharing the same `vercel.json` that lets us specify configuration
such as the rewrite needed to handle deep linking into the Ember UI. I
tried having Storybook use `/ui/storybook` as the root directory (and
adding a symbolically-linked `package.json` to bypass Vercel’s refusal
to build without it) but that produced broken Storybook deployments.

This instead combines the two projects into one
(`nomad-storybook-and-ui`), defaults to forwarding `/` to `/ui/`, and
adds the header link to the UI to navigate to Storybook.

Rather than have a complex build script in the Vercel configuration UI,
this delegates to a script in the repository.

image

Also of interest in the Vercel UI is this Git configuration that skips builds
when there are no changes within the root directory (`/ui/`).

image

### CODE REVIEW

File: ui/vercel.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "github": {
+    "silent": true
```
Comment (backspace) [inline ui/vercel.json:3]: This turns off comments like [this](https://github.com/hashicorp/nomad/pull/9471#issuecomment-735883596).

File: ui/vercel.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  },
+  "redirects": [
+    { "source": "/", "destination": "/ui/" }
+  ],
+  "rewrites": [
+    { "source": "/ui/(.*)", "destination": "/ui/index.html" }
+  ],
+  "trailingSlash": true
```
Comment (backspace) [inline ui/vercel.json:11]: This fixes a blank screen when visiting `/storybook`.

File: ui/.vercel/build.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+STORYBOOK_LINK=true ember build
+mkdir -p ui-dist/ui
+mv dist/* ui-dist/ui/
+
+yarn build-storybook
+mv storybook-static ui-dist/storybook/
```
Comment (DingoEatingFuzz) [inline ui/.vercel/build.sh:6]: I totally agree that this should live in the repo rather than the Vercel configuration UI 🙇

**Review by DingoEatingFuzz — APPROVED** [review]
Nicely done!

**Pull request merged**